x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_edit/threshold_rule_field_edit.tsx

```diff
@@ -10,19 +10,22 @@ import type { UpgradeableThresholdFields } from '../../../../model/prebuilt_rule
 import { KqlQueryEditForm } from './fields/kql_query';
 import { DataSourceEditForm } from './fields/data_source';
 import { ThresholdAlertSuppressionEditForm } from './fields/threshold_alert_suppression';
+import { ThresholdEditForm } from './fields/threshold/threshold_edit_form';
 
 interface ThresholdRuleFieldEditProps {
   fieldName: UpgradeableThresholdFields;
 }
 
 export function ThresholdRuleFieldEdit({ fieldName }: ThresholdRuleFieldEditProps) {
   switch (fieldName) {
-    case 'kql_query':
-      return <KqlQueryEditForm />;
-    case 'data_source':
-      return <DataSourceEditForm />;
     case 'alert_suppression':
       return <ThresholdAlertSuppressionEditForm />;
+    case 'data_source':
+      return <DataSourceEditForm />;
+    case 'kql_query':
+      return <KqlQueryEditForm />;
+    case 'threshold':
+      return <ThresholdEditForm />;
     default:
       return null; // Will be replaced with `assertUnreachable(fieldName)` once all fields are implemented
```

Comment:
What fields do we still need to implement here?

Reply:
Good catch! Added `assertUnreachable`.